actions/utils.go

```diff
@@ -46,3 +56,30 @@ func join(A []string, delim string) string {
 
 	return buffer.String()
 }
+
+// Randomly generate a set of indexes in each sector
+func generateInsertedIndexesForPearl(fileSizeInByte int) []int {
+	var indexes []int
+	if fileSizeInByte <= 0 {
+		return indexes
+	}
+
+	fileSectorInByte := fileChunkSizeInByte * (fileSectorInChunkSize - 1)
+	numOfSectors := int(math.Ceil(float64(fileSizeInByte) / float64(fileSectorInByte)))
+	remainderOfChunks := int(math.Ceil(float64(fileSizeInByte) / fileChunkSizeInByte))
+
+	for i := 0; i < numOfSectors; i++ {
+		rang := min(fileSectorInChunkSize, remainderOfChunks)
+		indexes = append(indexes, rand.Intn(rang))
+		remainderOfChunks = remainderOfChunks - (fileSectorInChunkSize - 1)
+	}
+	return indexes
+}
+
+// Util method to return min value of a and b.
```

Comment:
https://golang.org/pkg/math/#Min

Reply:
Correct, that would only apply for float64. Which mean I need to type cast it into float64 in this case.

Reply:
I see. golang makes me sad sometimes...

Reply:
You are not alone